### PR TITLE
feat: add secret discovery

### LINKED ISSUE #26: feature request: Automatically (try to) decrypt files listed in stack secrets
following #6 , it would be great to avoid having to list the encrypted files in the `sops_files` and having them automatically detected by reading the `secrets` part of the stack.

I understand this is an highly opinionated request, since it means that every secret file has to be encrypted with sops and set up to be decrypted with the SwarmCD age public key, but I believe it is a good constraint to have.

Nonetheless, the feature could be an opt-in and manageable at various level:

- `sops_secrets_discovery` option in `config.yaml` to enable the behavior globally;
- `sops_secrets_discovery` option in each stack of `/app/stacks.yml` to be able to override the global option
- keep the `sops_files` option if the previous options are set to false

What do you think?

### PR DESCRIPTION
**Pull request by sanzoghenzo**
this also adds some performance improvements by avoiding multiple I/O on files

closes #26

### CODE REVIEW

**Comment by sanzoghenzo** [comment]
I also have a fix for #53 - still have to try it so I'll wait until this PR is merged

**Review by m-adawi — APPROVED** [review]
Great work man. Thanks!

**Pull request merged**
